Jenkinsfiles/Jenkinsfile.deploy_ami

```diff
@@ -0,0 +1,216 @@
+      name: 'DEPLOY_BRANCH'
+    )
+    string(
+      defaultValue: "",
+      description: 'The Blue Button AMI id that will be used for the deployment.',
+      name: 'AMI_ID'
+    )
+    string(
+      defaultValue: "m3.medium",
+      description: 'The class/size of the ec2 instance to launch.',
+      name: 'INSTANCE_CLASS'
+    )
+    choice(
+      choices: 'dev\ntest\nimpl\nprod',
+      description: 'The environment to deploy to. Required.',
+      name: 'ENV'
+    )
+  }
+
+  stages {
+    stage('Ensure ENV and AMI_ID') {
+      steps {
+        sh """
+        if [ -z "${params.BRANCH}" ] || [ -z "${params.ENV}" ] || [ -z "${params.AMI_ID}" ] || [ -z "${params.SUBNET_ID}" ]
+        then
+          exit 1
+        fi
+        """
+      }
+    }
+
+    stage('Notify HipChat') {
+      steps {
+        withCredentials([
+          string(credentialsId: 'hipchat-room', variable: 'room'),
+          string(credentialsId: 'hipchat-server', variable: 'server'),
+          string(credentialsId: 'hipchat-token', variable: 'token')
+        ]) {
+          hipchatSend(
+            color: 'GRAY',
+            notify: true,
+            message: "STARTED: ${env.JOB_NAME} [${params.ENV}]",
+            room: room,
+            sendAs: '',
+            server: server,
+            token: token,
+            v2enabled: true
+          )
+        }
+      }
+    }
+
+    stage('Checkout') {
+      steps {
+        checkout([
+          $class: 'GitSCM',
+          branches: [[
+            name: "${params.DEPLOY_BRANCH}"
+          ]],
+          doGenerateSubmoduleConfigurations: false,
+          extensions: [[
+            $class: 'RelativeTargetDirectory',
+            relativeTargetDir: 'code'
+          ]],
+          userRemoteConfigs: [[
+            url: 'https://github.com/CMSgov/bluebutton-web-deployment.git'
+          ]]
+        ])
+      }
+    }
+
+    stage('Determine terraform config files') {
+      steps {
+        script {
+          if (params.ENV == 'dev') {
+            backend_config = 'bb-backend-dev'
+            terraform_vars = 'bb-tf-dev'
+          }
+          if (params.ENV == 'test') {
+            backend_config = 'bb-backend-test'
+            terraform_vars = 'bb-tf-test'
+          }
+          if (params.ENV == 'impl') {
+            backend_config = 'bb-backend-impl'
+            terraform_vars = 'bb-tf-impl'
+          }
+          if (params.ENV == 'prod') {
+            backend_config = 'bb-backend-prod'
+            terraform_vars = 'bb-tf-prod'
+          }
+        }
+      }
+    }
+
+    stage('Sanity check terraform plan') {
+      steps {
+        script {
+          dir('code') {
+            withAwsCli(credentialsId: aws_creds, defaultRegion: 'us-east-1') {
+              withCredentials([
+                file(credentialsId: backend_config, variable: 'bc'),
+                file(credentialsId: terraform_vars, variable: 'tv')
```

Comment:
These variables are a bit threaded through (pass to env to jenkins, jenkins sets them on a previous step based on env, then gets passed through here which turns it into an environment variable for the inline shell script).  I wonder if there's an easier way.  Perhaps the script can just search for a standard "bb-tf-${params.env}" terraform variables file?  This is related to the comment about taking the scripts out of the jenkins configuration as well.